src/client/datascience/jupyter/jupyterSession.ts

```diff
@@ -244,11 +244,12 @@ export class JupyterSession extends BaseJupyterSession {
         }
 
         // Create our session options using this temporary notebook and our connection info
-        const options: Session.IOptions = {
-            path: backingFile?.path || '.',
-            kernelName: getNameOfKernelConnection(kernelConnection) || '',
+        const options: Session.IOptions = { 
+            path: backingFile?.path || `${uuid()}.ipynb`, // Name has to be unique
```

Comment:
backingFile doesn't work on synapse (contents api is not supported), so this was always empty (which is an invalid file path).